### PR TITLE
alertmanager/data: Add tests

### PR DESCRIPTION
**Pull request by elad661**
This PR adds tests to the alertmanager package.

@nimrodshn @zgalor @jhernand @yaacov Please review.

### CODE REVIEW

**Comment by yaacov** [comment]
For bookkeeping :-)
https://github.com/openshift/autoheal/issues/44

File: pkg/alertmanager/data_test.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	a.Labels = map[string]string{"namespace": "bar"}
+	if a.Namespace() != "bar" {
+		t.Errorf("Expected bar but got %+v", a.Namespace())
+	}
+}
+
+func TestHash(t *testing.T) {
+	a := new(Alert)
+	a.Labels = map[string]string{"alertname": "foo", "test": "test"}
+	a.Annotations = map[string]string{"foo": "bar", "test1": "test"}
```
Comment (zgalor) [inline pkg/alertmanager/data_test.go]: This is really not that important but that way is more effective:

```
a := Alert{
  Labels: map[string][string]{
    "alertname": "foo",
    "test": "test",
  },
  Annotations: map[string][string]{
    "foo": "bar",
    "test1": "test",
  },
}
```

That way the struct fields get initialized with the values you want only once.
The way you wrote it, `Alert` is first initialized once with default empty maps and later overridden with the new values

Comment (nimrodshn) [reply pkg/alertmanager/data_test.go]: Also, I think using composite literals read better in my POV - I know we don't tend to use this approach throughout the project but the boiler-plating here scream 'use composite literals'.

For more information on `make` vs. `composite literals` vs `new` please refer to [this](https://golang.org/doc/effective_go.html#allocation_new).

**Comment by elad661** [comment]
@zgalor thanks, that does look more readable.

**Comment by nimrodshn** [comment]
Very nice. /lgtm

**Comment by nimrodshn** [comment]
@yaacov 
Thanks I edited my comment :+1:

**Comment by jhernand** [comment]
Thanks @elad661!

File: pkg/alertmanager/data_test.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+func TestNamespace(t *testing.T) {
+	a := new(Alert)
+	if a.Namespace() != "default" {
+		t.Errorf("Expected default but got %+v", a.Namespace())
+	}
+	a.Annotations = map[string]string{"namespace": "foo"}
+	if a.Namespace() != "foo" {
+		t.Errorf("Expected foo but got %+v", a.Namespace())
+	}
+	a.Labels = map[string]string{"namespace": "bar"}
```
Comment (zgalor) [inline pkg/alertmanager/data_test.go]: and here

Comment (zgalor) [reply pkg/alertmanager/data_test.go]: Ah, Now I understand what you did here.
Please ignore

File: pkg/alertmanager/data_test.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+package alertmanager
+
+import (
+	"testing"
+)
+
+func TestName(t *testing.T) {
+	a := new(Alert)
+	a.Labels = map[string]string{"alertname": "foo"}
```
[the remark this answers is not in the document]
Comment (zgalor) [reply pkg/alertmanager/data_test.go]: fixed here: https://github.com/openshift/autoheal/pull/81

**Pull request merged**